src/gitUtils.ts

```diff
@@ -37,13 +37,16 @@ export const pushTags = async () => {
 }
 
 export const switchToMaybeExistingBranch = async (branch: string) => {
-  const { stderr } = await execWithOutput('git', ['checkout', branch], {
+  const { stdout, stderr } = await execWithOutput('git', ['checkout', branch], {
```

Comment:
Is this a git version related issue? The source codes are copied from https://github.com/changesets/action/blob/master/src/gitUtils.ts#L39-L49, does this mean there are also same issue there?

Reply:
Definitely, the message that we are searching for is for `git checkout -b [branch]` but we are checking the output of `git checkout [branch]` here.

Reply:
If `git checkout` failed, it means the branch has not been created, so `git checkout -b` will be used later.

Reply:
I believe it should be something like:

```diff
 export const switchToMaybeExistingBranch = async (branch: string) => {
-  let { stderr } = await execWithOutput("git", ["checkout", branch], {
+  let { stdout } = await execWithOutput("git", ["checkout", branch], {
     ignoreReturnCode: true,
   });
-  let isCreatingBranch = !stderr
-    .toString()
-    .includes(`Switched to a new branch '${branch}'`);
+  let isCreatingBranch = !stdout.includes(`Switched to branch '${branch}'`);
   if (isCreatingBranch) {
     await exec("git", ["checkout", "-b", branch]);
   }
```

Reply:
Interesting. This was my first attempt to fix this issue!  
But it looks like the Switched to branch message is inside `stderr` instead of `stdout`. I think something is wrong with `execWithOutput` and I don't know what it is? So I went the safe path and checked both std streams 😄

Reply:
I tried, that's true. Then it should be `let isCreatingBranch = ! stderr.includes(`Switched to branch '${branch}'`);`